pkgs/development/python-modules/pytorch3d/default.nix

```diff
@@ -0,0 +1,50 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  setuptools,
+  wheel,
+  torch,
+  iopath,
+  cudaPackages,
+  config,
+  cudaSupport ? config.cudaSupport,
+}:
+
+assert cudaSupport -> torch.cudaSupport;
+
+buildPythonPackage rec {
+  pname = "pytorch3d";
+  version = "0.7.8";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "facebookresearch";
+    repo = "pytorch3d";
+    rev = "V${version}";
+    hash = "sha256-DEEWWfjwjuXGc0WQInDTmtnWSIDUifyByxdg7hpdHlo=";
+  };
+
+  nativeBuildInputs = lib.optionals cudaSupport [ cudaPackages.cuda_nvcc ];
+  build-system = [
+    setuptools
+    wheel
+  ];
+  dependencies = [
+    torch
+    iopath
+  ];
+  buildInputs = [ (lib.getOutput "cxxdev" torch) ];
```

Comment:
TIL

Reply:
Hopefully we can remove this once we switch to eval-time propagation of python `dependencies`...